src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py

```diff
@@ -0,0 +1,199 @@
+import os
+import logging
+import lxml.html as html
+from common.requester import DelayedRequester
+from common.storage.image import ImageStore
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+LIMIT = 35
+DELAY = 5.0
+RETRIES = 3
+PROVIDER = "brooklynmuseum"
+ENDPOINT = "https://www.brooklynmuseum.org/api/v2/object/"
+API_KEY = os.getenv("BROOKLYN_MUSEUM_API_KEY", "nokeyprovided")
+
+delay_request = DelayedRequester(delay=DELAY)
+image_store = ImageStore(provider=PROVIDER)
+
+HEADERS = {
+    "api_key": API_KEY
+}
+
+DEFAULT_QUERY_PARAM = {
+    "has_images": 1,
+    "rights_type_permissive": 1,
+    "limit": LIMIT,
+    "offset": 0
+}
+
+
+def main():
+    logger.info("Begin: Brooklyn museum provider script")
+    offset = 0
+    condition = True
+    while condition:
+        query_param = _get_query_param(offset)
+        objects_batch = _get_object_json(query_param=query_param)
+        logger.debug(len(objects_batch))
+        if len(objects_batch) > 0:
+            image_count = _process_objects_batch(objects_batch)
+            logger.debug(f"Images till now {image_count}")
+            offset += LIMIT
+        else:
+            condition = False
+    total_images = image_store.commit()
+    logger.info(f"Total images recieved {total_images}")
+
+
+def _get_query_param(offset=0,
+                     default_query_param=DEFAULT_QUERY_PARAM):
+    query_param = default_query_param.copy()
+    query_param.update(
+        offset=offset
+    )
+    return query_param
+
+
+def _get_object_json(headers=HEADERS,
+                     endpoint=ENDPOINT,
+                     retries=RETRIES,
+                     query_param=None):
+    for tries in range(retries):
+        response = delay_request.get(
+                    endpoint,
+                    query_param,
+                    headers=headers
+                    )
+        data = response.json()
```

Comment:
I wasn't paying close enough attention in our previous conversation about this part; sorry about that.  This line needs to be wrapped in a `try`, since some errors will end up with `response` being a `NoneType`, or at least not having a valid JSON body.

See the `get_response_json` function in the `DelayedRequester` class, as that's intended to become the canonical example.

Reply:
So , I thought that would be handled by the if condition 
``` 
data = response.json()
if data and data.get("message", "").lower() == "success.":
           response_json = data.get("data")
           break
else:
           response_json = None
```
Only if the JSON body has  ```message == "success"```, we would know its the right JSON body and take the data, rest all the cases its ignored.